benchmarks/spack/isambard3/grace/spack.yaml

```diff
@@ -146,9 +152,25 @@ spack:
         modules:
         - craype-network-ofi
         - cray-mpich/8.1.29
-    # cray-fftw:
-    #   externals:
-    #   - spec: cray-fftw@3.3.10.7
-    #     prefix: /opt/cray/pe/fftw/3.3.10.7/arm_grace/
-    #     modules:
-    #     - brics/cray-fftw/3.3.10.7
+    py-numpy:
+      externals:
+      - spec: py-numpy@1.24.4
+        modules:
+        - cray-python
+        prefix: /opt/cray/pe/python/3.11.7
+      buildable: false
+    py-scipy:
+      externals:
+      - spec: py-scipy@1.10.1
+        modules:
+        - cray-python
+        prefix: /opt/cray/pe/python/3.11.7
+      buildable: false
+    py-mpi4py:
+      externals:
+      - spec: py-mpi4py@3.1.4
+        modules:
+        - cray-python
+        prefix: /opt/cray/pe/python/3.11.7
+      buildable: false
```

Comment:
Any reason for using `buildable: false` for this packages?  Also, I believe you want also to append `^ python@3.11.7` to all these specs, to avoid making it match with other versions of python.

Reply:
I thought `buildable: false` guarantees that the external will always get used. Is it too strong requirement? Happy to remove it if you think that's better. Adding the python version to the spec

Reply:
`buildable: false` complicates environment resolution quite a bit whenever a package appears in the environment.  I'd expect troubles for `py-numpy` and `py-scipy` because they're somewhat common, `py-mpi4py` less so and I'd hope spack links it to the right MPI library anyway.